website/content/docs/app-config/scope.mdx

```diff
@@ -0,0 +1,133 @@
+---
+layout: docs
+page_title: Workspace and Label Scoping - Application Configuration
+description: |-
+  Waypoint allows you to set different configuration values for specific workspaces or labels. For example, you might set a database URL for staging different from production using workspaces, or you might use labels to set different addresses for different regions or any other metadata.
+---
+
+# Workspace and Label Scoping
+
+Waypoint allows you to set different configuration values for specific
+[workspaces](/docs/workspaces) or [labels](/docs/waypoint-hcl/variables/labels).
+For example, you might set a database URL for staging different from production
+using workspaces, or you might use labels to set different addresses for
+different regions or any other metadata.
+
+By default, Waypoint application configuration is exposed to the application
+in every scenario. By using the additional scoping constraints described in
+this document, you can limit what configuration values are set in what
+situations.
+
+## Workspace Scoping
+
+You can use workspace scoping to expose configuration only in specific,
+exact [workspaces](/docs/workspaces):
+
+```hcl
+config {
+  env = {
+    "DB_ADDR" = "dev.example.com"
+  }
+
+  workspace "production" {
+    env = {
+      "DB_ADDR" = "prod.example.com"
+    }
+  }
+}
+```
+
+In this example, the `DB_ADDR` environment variable will be set to
+"prod.example.com" only in the "production" workspace. Within the `workspace`
+stanza, you can use all available configuration parameters such as `env`,
+`internal`, `files`, etc.
+
+-> **Note:** Values aren't inherited from the global scope. For example, you
+cannot set an [internal variable](/docs/app-config/internal) at the root
+and access it within the `workspace` stanza. You must redeclare any internal
+variables used.
+
+### Non-Exact Workspace Matching
+
+The `workspace` stanza does not support non-exact workspace matching. However,
+you can use the `label` stanza documented below to do this. All workspaces
+automatically are exposed via the `waypoint/workspace` label and label
+selectors support
+[operators](https://www.consul.io/api-docs/features/filtering#matching-operators)
+such as `contains`, `matches`, etc. For example, you could match all workspaces
+that contain "dev" this way:
+
+```hcl
+config {
+  env = {
+    "DB_ADDR" = "dev.example.com"
+  }
+
+  label "waypoint/env contains \"dev\"" {
```

Comment:
Should this label be `waypoint/workspace`?

Reply:
Oy yeah it should!